Fix n_jobs for DBSCAN

JohanWork opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
Fixes #16299 
<!--
Fixes #16299 
-->


#### What does this implement/fix? Explain your changes.
The implementation updates documentation for sklearn.cluster.DBSCAN regarding n_jobs for precomputed distances

#### Any other comments?
My first commit to open source, happy for any feedback! 

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

Commit:
Adding comment about number of jobs not effecting if distances are precomputed

Commit:
Adding comment about number of jobs not effecting if distances are precomputed

Commit:
updating the text after feedback

Commit:
updating the text after feedback

rth commented on sklearn/cluster/_dbscan.py:
```diff
@@ -86,7 +86,9 @@ def dbscan(X, eps=0.5, min_samples=5, metric='minkowski', metric_params=None,
         ``-1`` means using all processors. See :term:`Glossary <n_jobs>`
-        for more details.
+        for more details. Parallel execution is implemented for computing the 
+        distances. If precomputed distance are used, parallel execution is not 
```
Thanks you. Probably better to  remove "Parallel execution is implemented for computing the distances." That's redundant with the first sentence in this paragraph. Also "If precomputed distance is used"? Otherwise LGTM.

JohanWork replied:
Thanks wil remove the redundancy. The ticket is about parallel execution not making a difference when precomputed distances are used and clarifying that in the documentation.  Thats way "If precomputed distance are used" is in there but should perhaps be updated?

rth replied:
Yes, sounds good.

DatenBiene committed:
DOC Update random_state entry for dummy / random_projection (#16347)

maikia committed:
EXA diabetes instead of Boston dataset for feature selection (#16305)

ogrisel committed:
DOC More explicit warnings about impurity based feat. imp. (#16382)

maikia committed:
DOC plot_partial_dependence_visualisation_api exchanged boston for diabetes dataset (#16341)

* exchanged the boston for diabetes dataset loading

* updated the feature names

* exchanged remaining feature names

* cleaning up

* changed features used to age and bmi for easier understanding

thomasjpfan committed:
TST Fixes test for California housing (#16389)

thomasjpfan committed:
MNT Adds filters to jinja template (#16133)

kno10 committed:
FIX  Elkan k-means does not stop if tol=0 (#16075)

jnothman committed:
ENH Perform KNN imputation without O(n^2) memory cost

Fixes #15604

This is more computationally expensive than the previous implementation,
but should reduce memory costs substantially in common use cases.

jnothman committed:
Revert "ENH Perform KNN imputation without O(n^2) memory cost"
accidentally pushed to master

This reverts commit ae9eaf83a123c033975d0f69a8217e6699951851.

rushabh-v committed:
ENH Add 'if_binary' option to drop argument of OneHotEncoder (#16245)

DatenBiene committed:
[DOC] Update random_state descriptions for mutual_info, unsupervised, .… (4) (#16320)

* Update random_state doc for mutual_info.py, unsupervised.py, testing.py and utils/init.py

* Update comments from pr

* Update sklearn/metrics/cluster/_unsupervised.py

Co-Authored-By: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

* Update sklearn/utils/__init__.py

Co-Authored-By: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

* Update sklearn/utils/__init__.py

Co-Authored-By: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

NicolasHug committed:
FIX  max_leaf_node and max_depth interaction in GBDT (#16183)

* fix max_leaf_node max_depth interaction

* Added test

* comment

* what's new

* simpler solution

* moved and simplified test

* typo

rushabh-v committed:
ENH Improve error message for sparse multilabel-indicator y in RandomForestClassifier (#15971)

adrinjalali committed:
MNT CI dropping python-3.5 (#15106)

* dropping python-3.5

* fix install guide

* fix setup.py

* fix circleci

* advanced_installation

* index.html

* readme

* pyparsing.py

* remove clean_warning_registry

* 1.13.1 and 1.19.1

* don't use 16.04

* 18.04 libatlas-dev -> libatlas-base-dev

* min pillow version for 3.6 is 4.2.1

* echo commands

* fix conflict, and 32 bit

* fix conflict for circle

* fix scikit-image version dep

* mark tests as xfail on 32bit py3.6

* move to 1.13.3 min numpy version, and simplify old code

* remove the rest of _object_dtype_isnan usages

* Revert "remove the rest of _object_dtype_isnan usages"

This reverts commit c6e867ea1fbb1b45fc8b4be1490c5e2736ad6892.

* fix issues raised by jeremy

* minor fix

* minor fixes mentioned by Olivier

jhlegarreta committed:
DOC: Mark the sentence end in `classification_report` (#16411)

jhlegarreta committed:
DOC Fix syntax in `model_evaluation` UG (#16410)

adrinjalali committed:
DOC promote shallow copy in the docs (#16423)

* DOC promote shallow copy in the docs

* a shorter version

jeremiedbb committed:
BLD Specify build time dependencies via pyproject.toml (#16244)

* add pyproject.toml

* do it in CI

* cln

* no build isolation in the doc

* bump dependencies

* no-build-isolation everywhere needed

adrinjalali committed:
MNT/CI install scikit-image if we test doc on azure (#15065)

adrinjalali committed:
DOC add Bunch to public docs and API (#16404)

* add Bunch to public docs and API

* address Thomas's suggestions

* use Thomas's description

oleksandr-pavlyk committed:
DOC Docstring example of classifier should import classifier (#16430)

edwardcqian committed:
DOC improve random state docstring in model_selection/split (#15575)

glemaitre committed:
DOC improve the documentation of OneHotEncoder for if_binary (#16428)

lopusz committed:
DOC follow doc formatting guideline for module gaussian_process (#16415)

chbrandt committed:
MNT Deprecate public attributes in SGD and PassiveAggresive classes (#16261)

johannfaouzi committed:
FEA Turn on early stopping in histogram GBDT by default (#14516)

ngonthier committed:
DOC Fix Tutorial example code lacking context #13566 (#14675)

* Add context to tutorial example

*  try to keep line length under 80 characters

* make CI pass

* MNT Skips unsupervised_learning when scikit-image is not installed

* DOC make code comments as text

* solve issue plotting

* fix

* center images

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Thomas J Fan <thomasjpfan@gmail.com>

jeremiedbb committed:
ENH Make NMF preserve floating dtype (#16280)

ogrisel committed:
MNT Update nightly build URL and release staging config (#16435)

corriebar committed:
DOC add examples for Gaussian Processes and improve User Guide (#15221)

thomasjpfan committed:
DOC Improves readability of kernel functions in SVM docs (#16440)

ogrisel committed:
DOC Improve documentation for permutation importance (#16402)

ariepratama committed:
API Deprecates the fit_grid_point function (#16401)

VarIr committed:
TST Fix unreachable code in tests (#16110)

Commit:
ENH Sample weights for ElasticNet (#15436)

simonamaggio committed:
BUG raise value error when distance matrix is not square and affinity=precomputed (#16257)

glemaitre committed:
MAINT split partial dependence and visualization (#16453)

maikia committed:
DOC plot voting regression now uses diabetes dataset (#16387)

thomasjpfan committed:
BUG Fixes benchmark script for early stopping (#16476)

venkyyuvy committed:
ENH adding fit_params option for RegressorChain (#16111)

cmarmo committed:
DOC Fix new sphinx warnings (#16483)

jeremiedbb committed:
BUG accept all integer types as n_clusters in Birch (#16484)

narendramukherjee committed:
ENH IterativeImputer support array-like max and min (#16403)

adrinjalali committed:
API make a few estimators' inits kw-only (#16474)

chofchof committed:
FIX _get_estimators_indices() in BaseBagging (#16437)

jeremiedbb committed:
[MRG] new K-means implementation for improved performances (#11950)

lopusz committed:
DOC improve docstring following doc guide in module compose (#16416)

rth committed:
ENH XFAIL in common tests with estimator tags (#16502)

Commit:
[MRG] Document bounds='fixed' in GP kernels  (#16209)

jeremiedbb committed:
MAINT Clean root of sklearn repo (#16485)

sergulaydore committed:
TST check sample_weight shape added to common tests (#11598)

thomasjpfan committed:
TST Sets random state in test_csr_row_norms (#16509)

thomasjpfan committed:
ENH Improves efficiency of ColumnTransformer for string keys (#16431)

thomasjpfan committed:
DOC Fixes link in column transformer (#16516)

Commit:
DOC remove backticks in plain text in developer guide (#16521)

parsons-kyle-89 committed:
BUG fix behaviour in confusion_matrix with with empty array-like as input (#16442)

glemaitre committed:
BUG ensure that estimator_name is properly stored in the ROC display (#16500)

glemaitre committed:
BUG ensure that name is properly stored in the precision/recall display (#16505)

glemaitre committed:
DOC use make_column_selector inside the mixed type example (#16472)

jnothman committed:
ENH Perform KNN imputation without O(n^2) memory cost (#16397)

NicolasHug committed:
ENH Fast PDP for DecisionTreeRegressor and RandomForestRegressor (#15864)

NicolasHug committed:
MNT Removed out of date note about KMeans parallelism on macOS (#16527)

adrinjalali committed:
ENH Support sample weights in HGBT (#14696)

thomasjpfan committed:
TST Checks can now skip test based on estimator tag _xfail_test (#16510)

plgreenLIRU committed:
FIX Predicted standard deviation values of Gaussian Processes are only within [0, 1] (#15782)

* Initial work for WIP pull request

* First commit of full solution and new tests

* Changes from PR review

* Updated changelog and normalize_y description

* Updating v0.23.rst with upstream master

* Small update to whatsnew

Co-authored-by: Joel Nothman <joel.nothman@gmail.com>

glemaitre committed:
BUG avoid nan variance with sparse input in StandardScaler (#16466)

rth committed:
DOC Note on commit co-authorship when merging PRs (#16550)

CastaChick committed:
DOC Fixed documents that refer to Bunch object #16438 (#16447)

* Added links to utils.Bunch and fixed format of the docstring in datasets

* Added links to utils.Bunch in sklearn.compose

* Added links to utils.Bunch in sklearn.tree

* Added links to utils.Bunch in sklearn.ensemble

* Added links to utils.Bunch in sklearn.inspection

* Added links to utils.Bunch in sklearn.pipeline

* modified docstring of Bunch

* Added links to utils.Bunch to index.rst of sklearn.datasets

* Fixed some docstrings because the lines are too long

* Fixed some points as reviewed.

* Add links and delete 'for more information...'

* Fixed indent

* Fixed forgotten points.

* Fixed some points as reviewed.

thomasjpfan committed:
MNT Adds autolabler for modules (#16520)

* MNT Adds labeler

* BUG Fix

* Double quotes are better

* BUG Fix

* MNT Adds build ci tag

* MNT Use fork for new feature

adrinjalali committed:
API make cluster's __init__ parameters kwonly (#16501)

thomasjpfan committed:
MNT Fix autolabeler by pinning the version (#16566)

rth committed:
MAINT Use "Bug: triage" tag for user reports (#16560)

thomasjpfan committed:
MNT Removes autolabeler (#16577)

cmarmo committed:
ENH Include verbose message when sample_weight is provided. (#16564)

arunavkonwar committed:
DOC Remove 3.5 from Python badge in README (#16583)

NicolasHug committed:
DOC Updated gradient boosting UG (#16178)

* Updated gradient boosting UG

* maybe fix sphinx warnings

* Addressed comments from Guillaume

jeremiedbb committed:
move 0.22.2 what's new entries (#16586)

jeremiedbb committed:
DOC remind to update the version in maintainers guide (#16589)

amueller commented:
Hey @JohanWork. Are you still interested in following up on this? I think it's nearly ready to go!

NicolasHug committed:
MNT Introduction of n_features_in_ attr with _validate_data mtd (#16112)

jumon committed:
DOC Rename clf to regr in SVR examples (#16598)

timnon committed:
DOC Adds example for RandomTreesEmbedding (#15202)

Commit:
MNT rename _parallel_fit_estimator to _fit_single_estimator to reflect lack of parallelism in the method (#16599)

pspachtholz committed:
DOC Adds examples to GradientBoostingClassifier and GradientBoostingRegressor (#15151)

Rick-Mackenbach committed:
Fix format of values in confusion matrix plot. (#16159)

Shiki-H committed:
DOC Add formula for binary balanced accuracy in UG (#16604)

JohanWork commented:
@amueller Hi sorry! So that @hugolmn approved but @rth  do you still want the update if so, I will fixit tomorrow! Sorry for being slow in the response, will improve!

rth commented:
> do you still want the update if so, I will fixit tomorrow! 

Yes, please. Thanks!

rth commented:
Also flake8 linting is failing (probably too long line, see the logs).

VarIr committed:
MNT Download and test datasets in cron job (#16348)

FollonSaxBass committed:
DOC Added MLPRegressor and MLPClassifier examples (#15228)

* Added MLPRegressor and MLPClassifier examples

* shortened line due to test failure

* changed way to calculate score due to approximation error in tests

* removed comments, used predict instead of cross validation

* DOC Simplified make_regression and make_classification arguments

* DOC fix for linting

* DOC Update

* DOC Less precision

Co-authored-by: Thomas J Fan <thomasjpfan@gmail.com>

thomasjpfan committed:
MNT Periodic adds labels based on module (#16596)

* MNT Users pull request labeler

* MNT Fix

* labeler fix

* MNT Update

* MNT Uses single quotes

* BUG Uses number

* MN Removes Build / CI auto labeling

* MNT Updates to labeler v2.4.1

* ENH Updates version to include logs

Commit:
updating after feedback

Commit:
updating after feedback

Commit:
update linting

Commit:
update linting

Commit:
update for linting

Commit:
update for linting

Commit:
removing spaces

Commit:
removing spaces

Commit:
Merge branch 'nJobs' of github.com:JohanWork/scikit-learn into nJobs

JohanWork commented:
I made a misstake, trying to rebase.
